spec/ddtrace/contrib/graphql/tracer_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+require 'ddtrace/contrib/graphql/test_types'
+
+require 'ddtrace'
+
+# rubocop:disable Metrics/BlockLength
+RSpec.describe 'GraphQL patcher' do
+  include_context 'GraphQL test schema'
+
+  # GraphQL generates tons of warnings.
+  # This suppresses those warnings. 
+  around(:each) do |example|
+    without_warnings do
+      example.run
+    end
+  end
+
+  let(:tracer) { ::Datadog::Tracer.new(writer: FauxWriter.new) }
+
+  def all_spans
+    tracer.writer.spans(:keep)
+  end
+
+  before(:each) do
+    Datadog.configure do |c|
+      c.use :graphql,
+            service_name: 'graphql-test',
+            tracer: tracer,
+            schemas: [schema]
+    end
+  end
+
+  describe 'query trace' do
+    subject(:result) { schema.execute(query, variables: {}, context: {}, operation_name: nil) }
+
+    let(:query) { '{ foo(id: 1) { name } }' }
+    let(:variables) { {} }
+    let(:context) { {} }
+    let(:operation_name) { nil }
+
+    it do
+      # Expect no errors
+      expect(result.to_h['errors']).to be nil
+
+      # Expect nine spans
+      expect(all_spans).to have(9).items
+
+      # Expect each span to be properly named
+      all_spans.each do |span|
+        expect(span.service).to eq('graphql-test')
+        expect(span.resource.to_s).to_not be_empty
```

Comment:
what kind of resource are we testing here? I would prefer to be 100% sure that we expect some values rather than "it's not empty". Especially because `::GraphQL::Tracing::DataDogTracing` is not maintained in this library and so would be great knowing about breaking changes.

Reply:
This used to be the broken thing in `graphql`. It used to be empty, now it returns something. We could update that assertion to be a bit more explicit, but it's going to be different for every single span. Since `graphql` is doing the instrumentation here, if they change what spans they record, it'll break our CI suite.

We probably shouldn't be asserting specific side effects from GraphQL at all, only that we were able to get some spans that look like GraphQL is working still. Otherwise we could find ourselves having to fix a very brittle test anytime they change something on their end.